### PR TITLE
Help Center: Update transfer to support

### PR DESCRIPTION
**Pull request by agrullon95**
Related to https://github.com/Automattic/dotcom-forge/issues/9746

## Proposed Changes

* Update header text to `Support Team` when chatting with support.
* DIsplay `Chatting with support` message before the user is transferred to support chat.
* Update AI message before transferring the user to Support.

CleanShot 2024-11-14 at 17 40 59@2x

https://github.com/user-attachments/assets/90dd4ca3-3017-46b8-9f91-dd8ef8e82098

## Why are these changes being made?

* Update experience and improve clarity when the user is transferred from AI chat to Support chat.

## Testing Instructions

* Checkout branch or use live link.
* Navigate to `/home?flags=help-center-experience`
* Open a new chat and verify the header text is `Support Assistant`
* Send a message `Support` and click the `Get instant support` from the AI response.
* Verify the header changes to `Support Team` and the `Chatting with support now` text appears above the initial support message.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/help-center/src/components/help-center-header.tsx [hunk 1]
```diff
@@ -94,22 +96,39 @@ const ChatEllipsisMenu = () => {
...
+				return foundMatchingSupportConversation
+					? __( 'Support Team', __i18n_text_domain__ )
+					: __( 'Support Assistant', __i18n_text_domain__ );
+			}
+			return __( 'Wapuu', __i18n_text_domain__ );
+		};
+
 		switch ( pathname ) {
+			case '/odie':
+				return odieHeader();
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-header.tsx]: Can we call this `getOdieHeader`?

File: packages/odie-client/src/constants.ts [hunk 2]
```diff
@@ -66,6 +66,61 @@ export const ODIE_TRANSFER_MESSAGE = (
 	},
 } );
 
+export const ODIE_TRANSFER_MESSAGE_NEW = (
```
Comment (escapemanuele) [inline packages/odie-client/src/constants.ts]: @agrullon95 can we do something like we did for `getOdieInitialMessageConstant`? Create a function to return the right constant?

Comment (escapemanuele) [reply packages/odie-client/src/constants.ts]: I don't particularly like the idea of transforming constants in functions :)

File: packages/help-center/src/components/help-center-header.tsx (hunk 1 above)
Comment (agrullon95) [reply packages/help-center/src/components/help-center-header.tsx]: Updated to `getOdieHeader`

File: packages/odie-client/src/constants.ts [hunk 3]
```diff
@@ -49,76 +49,59 @@ export const ODIE_FORWARD_TO_ZENDESK_MESSAGE = __(
...
-					flags: {
-						hide_disclaimer_content: true,
-						show_contact_support_msg: false,
-						show_ai_avatar: false,
-					},
-					site_id: null,
-				},
+export const ODIE_TRANSFER_MESSAGE_NEW: Message[] = [
+	{
+		content: __( "We're connecting you to our support team.", __i18n_text_domain__ ),
```
Comment (escapemanuele) [inline packages/odie-client/src/constants.ts]: Mmm before it was like

```
content: shouldUseHelpCenterExperience
		? __( "Help's on the way!", __i18n_text_domain__ )
```

I believe this should be the correct text.

Sorry again, the screenshot maybe was misleading. The copy is in the doc 1ptULS7zslhT8vPZEL3hD5CMwv2hALByQB36hMhzQwKI-gdoc

To make this easier, copy as it was before is correct

Comment (escapemanuele) [reply packages/odie-client/src/constants.ts]:
```
New copy: Help’s on the way! 
Feel free to share more details while we connect you to support. 
```

Comment (agrullon95) [reply packages/odie-client/src/constants.ts]: Thanks for the updated copy. I updated my PR to address this.

**Review by escapemanuele — APPROVED** [review]
Pushed some simplification, please give it a try ✌️

**Comment by agrullon95** [comment]
@escapemanuele - Thanks for the changes. Everything is still working. I will merge and open another PR for more header related changes.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17003990).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17003990).

**Pull request merged**